File: aq_backend/lambda/risk_score.py
def personal_risk_score(user):
    """
    Calculate the personal risk score based on the user's health and activity information.
    The input `user` should be a dict with keys matching the form fields.
    Returns an integer risk score (0-15).
    """
    risk_score = 0
    if user.get('ageGroup') in ['0-4', '65+']:
        risk_score += 2
    elif user.get('ageGroup') == '5-17':
        risk_score += 1
    if user.get('hasLungCondition') == 'yes':
        risk_score += 3
    if user.get('isPregnant') == 'yes':
        risk_score += 2
    if user.get('hasHeartCondition') == 'yes':
        risk_score += 2
    if user.get('outdoorsMoreThanTwoHours') == 'yes':
        risk_score += 1
    if user.get('activityIntensity') == 'hard':
        risk_score += 2
    elif user.get('activityIntensity') == 'moderate':
        risk_score += 1
    if user.get('workSetting') == 'outdoor':
        risk_score += 2
    elif user.get('workSetting') == 'indoor-windows':
        risk_score += 1
    if user.get('smokesOrVapes') == 'yes':
        risk_score += 2
    if user.get('hasAirPurifier') == 'yes':
        risk_score -= 1
    # Clamp to [0, 15]
    risk_score = max(0, min(risk_score, 15))
    return risk_score

File: aq_app/src/components/riskScore.js
// Utility to calculate personal risk score from form data
export function calculateRiskScore(formData) {
  let riskScore = 0;
  if (formData.ageGroup === '0-4' || formData.ageGroup === '65+') riskScore += 2;
  if (formData.ageGroup === '5-17') riskScore += 1;
  if (formData.hasLungCondition === 'yes') riskScore += 3;
  if (formData.isPregnant === 'yes') riskScore += 2;
  if (formData.hasHeartCondition === 'yes') riskScore += 2;
  if (formData.outdoorsMoreThanTwoHours === 'yes') riskScore += 1;
  if (formData.activityIntensity === 'hard') riskScore += 2;
  if (formData.activityIntensity === 'moderate') riskScore += 1;
  if (formData.workSetting === 'outdoor') riskScore += 2;
  if (formData.workSetting === 'indoor-windows') riskScore += 1;
  if (formData.smokesOrVapes === 'yes') riskScore += 2;
  if (formData.hasAirPurifier === 'yes') riskScore -= 1;
  // Clamp min to 0
  const minScore = 0;
  const maxScore = 15;
  // Clamp riskScore to [0, maxScore]
  const rawScore = Math.max(minScore, Math.min(riskScore, maxScore));
  const normalizedScore = Math.round((rawScore / maxScore) * 100);
  let invertedScore = 100 - normalizedScore;
  // Clamp final score to [0, 100]
  invertedScore = Math.max(0, Math.min(invertedScore, 100));
  return { rawScore, normalizedScore, invertedScore };
}
